core/extension/gdextension_interface.h

```diff
@@ -386,6 +388,8 @@ typedef void *GDExtensionScriptInstancePtr; // Pointer to ScriptInstance.
 typedef struct {
 	GDExtensionScriptInstanceSet set_func;
 	GDExtensionScriptInstanceGet get_func;
+	GDExtensionScriptInstanceGetClassCategory get_class_category_func;
+	GDExtensionScriptInstanceFreeClassCategory free_class_category_func;
```

Comment:
Adding functions into an existing struct will created backward compatibility issues.

I think we'd need to make a new struct with the new properties added. And then add a new version of the "script_instance_create" function (maybe just "script_instance_create2"?) that takes the new struct. This way, existing GDExtensions would continue to work without being recompiled, but new GDExtensions would be able call the new function with the new struct.